packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -311,14 +311,12 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
 
   private _onRenderTextContents = (): JSX.Element | (JSX.Element | null)[] => {
     const {
-      text,
-      children,
       secondaryText = this.props.description,
       onRenderText = this._onRenderText,
       onRenderDescription = this._onRenderDescription
     } = this.props;
 
-    if (text || typeof (children) === 'string' || secondaryText) {
+    if (this._hasText() || secondaryText) {
```

Comment:
Wanted to point out that the refactor actually made this a stricter check. Before, if text was null and typeof (children) === 'string' evaluated to true, the if was true. Now, text must be undefined for typeof (children) check to make the if truthy. I think this was the original intention because falling back to string children for text is supposed to be for backward compatibility and assumes text property is not present, but it's still a change that I want scrutinized by reviewers.

Reply:
While I agree that this seems safe, let's not change the logic because it may have a ripple effect for some consumers. Feel free to log a bug and list it in a comment inside of _hasText so that it can be addressed later

Reply:
Reverted to reduce risk

---
In reply to: [190656224](https://github.com/OfficeDev/office-ui-fabric-react/pull/4963#discussion_r190656224) [](ancestors = 190656224)